testsuite/tests/parallel/catch_break.ml

```diff
@@ -23,43 +44,65 @@ Success.
 
 *)
 
+let delay = 0.001 (* 1 ms *)
+let fuel = Atomic.make 1000 (* 1s *)
+
 let print = if verbose then print_endline else fun _ -> ()
 
+let ready_count = Atomic.make 0
+
+(* Does not poll *)
+
+let sleep () =
+  if Atomic.get fuel <= 0 then (
+    print "[Reached max attempts without succeeding]";
+    Unix._exit 1
+  );
+  Atomic.decr fuel;
+  Unix.sleepf delay
+
+let rec wait n =
+  if Atomic.get ready_count <> n then (
+    sleep ();
+    wait n
+  )
```

Comment:
It is not obvious to me why the test performs synchronization between the sender and the receiver by using busy-waiting rather than a proper synchronization mechanism, such as a mutex. Can you maybe explain this choice in a comment?